src/proxy-ui-api/frontend/src/views/Diagnostics/Diagnostics.vue

```diff
@@ -1,16 +1,216 @@
+                <th>{{$t('diagnostics.message')}}</th>
+                <th class="time-column">{{$t('diagnostics.previousUpdate')}}</th>
+                <th class="time-column">{{$t('diagnostics.nextUpdate')}}</th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr v-for="ocsp in ocspDiags.ocsp_responders" v-bind:key="ocsp.url">
+                <td>
+                  <StatusIcon :status="statusIconType(ocsp.status_class)" />
+                </td>
+                <td class="url-column" data-test="service-url">{{ocsp.url}}</td>
+                <td>{{$t('diagnostics.ocspResponders.ocspStatus.'+ocsp.status_code)}}</td>
+                <td class="time-column">{{ocsp.prev_update_at | formatHoursMins}}</td>
+                <td class="time-column">{{ocsp.next_update_at | formatHoursMins}}</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </v-card-text>
+    </v-card>
   </v-layout>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
+import * as api from '@/util/api';
+import {
+  TimestampingServiceDiagnostics,
+  OcspResponderDiagnostics,
+  GlobalConfDiagnostics,
+} from '@/types';
+import StatusIcon from '@/components/ui/StatusIcon.vue';
 
 export default Vue.extend({
-  data: () => ({}),
+  components: {
+    StatusIcon,
+  },
+  data: () => ({
+    timestapmingServices: [] as TimestampingServiceDiagnostics[],
+    globalconf: undefined as GlobalConfDiagnostics | undefined,
+    ocspResponderDiagnostics: [] as OcspResponderDiagnostics[],
+  }),
+  methods: {
+    fetchData(): void {
+      api
+        .get(`/diagnostics/timestamping-services`)
+        .then((res) => {
+          this.timestapmingServices = res.data;
+        })
+        .catch((error) => {
+          this.$store.dispatch('showError', error);
+        });
+
+      api
+        .get(`/diagnostics/globalconf`)
+        .then((res) => {
+          this.globalconf = res.data;
+        })
+        .catch((error) => {
+          this.$store.dispatch('showError', error);
+        });
+
+      api
+        .get(`/diagnostics/ocsp-responders`)
+        .then((res) => {
+          this.ocspResponderDiagnostics = res.data;
+        })
+        .catch((error) => {
+          this.$store.dispatch('showError', error);
+        });
+    },
+
+    statusIconType(status: string): string {
+      if (!status) {
+        return '';
+      }
+      switch (status) {
+        case 'OK':
+          return 'green';
+        case 'WAITING':
+          return 'orange-ring';
+        case 'FAIL':
+          return 'red';
+        default:
+          return 'red';
+      }
+    },
+  },
+  created() {
+    this.fetchData();
+  },
 });
 </script>
 
 <style lang="scss" scoped>
+@import '../../assets/shared';
+@import '../../assets/colors';
+@import '../../assets/tables';
+
+.diagnostic-card {
+  &:first-of-type {
+    margin-top: 40px;
+  }
+
+  width: 100%;
```

Comment:
Preferably group all the style rules of a single element to same place. This width: 100%; and margin-bottom: 30px; below should stay next to each other so it'll be easier to see what rules are set for this class.

Reply:
Fixed